assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentService.java

```diff
@@ -817,6 +818,16 @@ public String getDeepLinkWithPermissions(String context, String assignmentId, bo
      */
     public List<MultiGroupRecord> checkSubmissionForUsersInMultipleGroups(String siteId, Group submissionGroup, Collection<Group> asnGroups);
 
+    public boolean correctTime(String timeSheet);
+
+    public AssignmentSubmissionSubmitter getSubmissionSubmitter(String submitterId) throws PermissionException;
+
+    public AssignmentTimeSheet getTimeSheet(String timeSheetId) throws PermissionException;
+	
+    public void addAssignmentTimeSheet(AssignmentTimeSheet timeSheet, String context) throws PermissionException;
+
+    public void removeAssignmentTimeSheet(AssignmentTimeSheet timeSheet, String context) throws PermissionException;
```

Comment:
if you know the timesheet why do you need the context here?

Reply:
To perform a permission check and make sure the record can be deleted

Implementation:
`    public void removeAssignmentTimeSheet( ... ) throws PermissionException {
        if (!allowAddSubmission(siteId)) {
            throw new PermissionException(sessionManager.getCurrentSessionUserId(), SECURE_ADD_TIMESHEET, null);
        }
        assignmentRepository.deleteAssignmentTimeSheet(timeSheet);
        log.debug("Remove timeSheet {}", timeSheet.getId());
    }`